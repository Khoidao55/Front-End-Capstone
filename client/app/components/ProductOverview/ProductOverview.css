.product-overview-container {
  display:flex;
  /* background-color: yellow; */
  height: 900px;
  margin-left: 20%;
}

.product-image-container {
  display: flex;
  /* background-color: grey; */
  flex-direction: column;
  justify-content: center;
  width: 900px;
  height: 700px;
  margin-left: -150px;
  /* border-style: solid; */
}

.product-info-container {
  display: flex;
  flex-direction: column;
  /* background-color: grey;
  border-style: solid; */
  margin: 10px 10px;
  height: 700px;
  /* width: 500px; */
  /* margin-left: 60px; */
}

.product-description-container {
  display: flex;
  /* background-color: red; */
  margin-top: 20px;
}

.product-style-container {
  display: flex;
  flex-direction: row;
  width: 340px;
  /* background-color: red; */
  margin-top: 20px;
}

.product-cart-container {
  display: flex;
  margin-top: 20px;
}

.product-summary-container {
  display: block;
  position: absolute;
  margin-top: 720px;
  margin-left: -100px;
  /* width: 100%; */
  /* background-color: grey;
  border-style: solid; */
}

.image-right-btn-active {
  display: flex;
  position: absolute;
  margin-left: 800px;
}

.image-right-btn-disabled {
  display: none;
}

.image-left-btn-active {
  display: flex;
  position: absolute;
  z-index: 19;
}

.image-left-btn-disabled {
  display: none;
}