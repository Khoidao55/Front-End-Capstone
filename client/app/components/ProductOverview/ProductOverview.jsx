import React, { useState, useEffect } from 'react';
import propTypes from 'proptypes';
import ProductCategoryAndTitle from './ProductCategoryAndTitle/ProductCategoryAndTitle.jsx';
import ProductImageGallery from './ProductImageGallery/ProductImageGallery.jsx';
import ProductInfo from './ProductInfo/ProductInfo.jsx';
import ProductStyleSelector from './ProductStyleSelector/ProductStyleSelector.jsx';
import ProductCart from './ProductCart/ProductCart.jsx';
import './ProductOverview.css';

const ProductOverview = ({ product, styles }) => {
  if (styles !== undefined) {
    const [priceOfProduct, setPriceOfProduct] = useState(product.originalPrice);
    const [selectedStyle, setSelectedStyle] = useState(styles[0]);

    useEffect(() => {
      setPriceOfProduct(() => product.originalPrice);
      setSelectedStyle(() => styles[0]);
    }, [product, styles]);

    const handleSelectedStyle = (style, price) => {
      setSelectedStyle(style);
      setPriceOfProduct(price);
    };

    return (
      <div className="product-overview-container">
        <div className="product-image-container">
          <ProductImageGallery style={selectedStyle} />
        </div>
        <div className="product-info-container">
          <div className="product-description-container">
            <ProductCategoryAndTitle price={priceOfProduct} product={product} />
          </div>
          <div className="product-style-container">
            <ProductStyleSelector
              handleSelectedStyleClick={handleSelectedStyle}
              styles={styles}
              styleName={selectedStyle}
            />
          </div>
          <div className="product-cart-container">
            <ProductCart />
          </div>
        </div>
        <div className="product-summary-container">
          <ProductInfo product={product} />
        </div>
      </div>
    );
  }

  return (
    <div className="product-overview-container">
      <div className="product-image-container" />
      <div className="product-info-container">
        <div className="product-description-container" />
        <div className="product-style-container" />
        <div className="product-cart-container" />
      </div>
      <div className="product-summary-container" />
    </div>
  );
};

ProductOverview.propTypes = {
  product: propTypes.object,
  styles: propTypes.array,
};

export default ProductOverview;
