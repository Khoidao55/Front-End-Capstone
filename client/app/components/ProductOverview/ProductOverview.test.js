import React from 'react';
import { shallow } from 'enzyme';
import ProductOverview from './ProductOverview.jsx';

describe('Product Overview', () => {
  test('should render my component', () => {
    const wrapper = shallow(<ProductOverview />);

    expect(wrapper).toMatchSnapshot();
  });
});
