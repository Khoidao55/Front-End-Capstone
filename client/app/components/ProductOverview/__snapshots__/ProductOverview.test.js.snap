// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Overview should render my component 1`] = `ShallowWrapper {}`;
